zerver/views/auth.py

```diff
@@ -602,11 +714,19 @@ def dev_direct_login(request: HttpRequest, **kwargs: Any) -> HttpResponse:
     if user_profile is None:
         return HttpResponseRedirect(reverse('dev_not_supported'))
     do_login(request, user_profile)
+    if settings.TWO_FACTOR_AUTHENTICATION_ENABLED:
+        # Login with two factor authentication as well.
+        do_two_factor_login(request, user_profile)
```

Comment:
Can you explain why this is in addition to `do_login`, rather than replacing it?

Reply:
Merged in `do_login`.